specs/vep-1416-vdk-troubleshooting-tools/README.md

```diff
@@ -0,0 +1,100 @@
+- Data Job: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job
+- Data Job Execution: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job-execution
+- Data Job Deployment: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job-deployment
+- Kubernetes: https://kubernetes.io/
+- Prometheus: https://prometheus.io/docs/introduction/overview/
+
+
+## Motivation
+
+-----
+When a data job, which is deployed in a kubernetes cluster, fails with an error, the VDK Deployment administrator can troubleshoot it either through the exposed Prometheus metrics or by looking at the job's logs. This works in most cases, but fails when there are issues with the logging service, some dependency does not propagate its logs, or when the configuration of the environment has been changed.
+
+In such cases, there are a couple of approaches that can be taken:
+
+1. An execution of the job can be triggered manually, and the administrator can connect to the cluster directly and either examine the logs as they are produced, or attach to the pod of the data job and troubleshoot it remotely. This has the limitation that the person doing the troubleshooting needs to have access to the kubernetes cluster, and even then the container in which the data job is running may not have all the utilities (editor, program for monitoring of running processes, etc.) to allow for proper troubleshooting. Additionally, it may not be possible for the data job to be run off its schedule due to business concerns, in which case the users/admins would need to wait for the job's next scheduled execution.
+2. The data job can be executed locally, so that the user/admin can have complete control over the environment. This again has the drawback that off-schedule execution of the data job may not be possible. Also, in case the issue is with some environment configuration or system dependency, the error may not be reproducible locally, which can make troubleshooting even more difficult.
+
+To allow for easier troubleshooting of errors with data jobs, a special plugin will be introduced, which will provide capabilities to do a thread dump and either log the contents or stored locally, where it can be examined by the owner of the data job or by the team responsible for the VDK deployment. The plugin could also be extended in the future with other debugging capabilities.
+
+
+## Requirements and goals
+
+-----
+### Goals
+
+* **Introduce a vdk-jobs-troubleshooting plugin**
+
+* **Provide capability to do a thread dump**
+  - A user/admin wants to get a thread dump of the data job process for troubleshooting purposes. For example, a user deploys a data job and after some time the job starts failing without any apparent reason. The user needs to be able to get a thread dump, so they are able to properly investigate the root cause for the issue.
+
+### Non-Goals
+
+* **Additional troubleshooting tools are not planned as part of this proposal.**
```

Comment:
Based on security section - the idea is for this feature to be exposed only for administrators of the VDK deployment and not end users (data engineers - data job developers). 
Which is fine . It makes the design simpler since security implications are different. 

But let's put it as non goal explicitly. My first thought was that end users would be able to use it.

Reply:
Well, if the data is send to the logging service that the VDK deployment uses, then end-users would, in theory, be able to see the logs (depending on what permissions they have to view logs).

Reply:
Updated the non-goals section. Thanks